Common/util/stream.cpp

```diff
@@ -107,12 +111,42 @@ size_t Stream::WriteByteCount(uint8_t b, size_t count)
     return size;
 }
 
+//-----------------------------------------------------------------------------
+// StreamSection
+//-----------------------------------------------------------------------------
+
+StreamSection::StreamSection(std::unique_ptr<IStreamBase> &&base, soff_t start, soff_t end)
+{
+    OpenSection(base.get(), start, end);
+    _ownBase = std::move(base);
+    _base = _ownBase.get();
+}
 
 StreamSection::StreamSection(IStreamBase *base, soff_t start, soff_t end)
-    : _base(base)
 {
-    _start = std::max((soff_t)0, std::min(start, end));
-    _end = std::min(std::max((soff_t)0, end), base->GetLength());
+    OpenSection(base, start, end);
+    _base = base;
+}
+
+void StreamSection::Open(IStreamBase *base)
+{
+    if (!base)
+        throw std::runtime_error("Base stream invalid.");
```

Comment:
Hey, can you try adding `#include <stdexcept>` on top?

I gave a try [here](https://github.com/ericoporto/ags/commit/97e16415a2993c886f6ae7c9478b1b60ac8450af) and it fixed the build errors [in my GHA CI](https://github.com/ericoporto/ags/actions/runs/8445607277)